### PR TITLE
feat(Zybooks): add zybooks

### PR DESCRIPTION
**Pull request by PythonCoderAS**
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

Library:
Screenshot 2023-09-15 at 6 48 31 PM
Zybook:
Screenshot 2023-09-15 at 6 50 31 PM
Assignments:
Screenshot 2023-09-15 at 6 48 45 PM
Screenshot 2023-09-15 at 6 49 04 PM
Section:
Screenshot 2023-09-15 at 6 49 10 PM
Screenshot 2023-09-15 at 6 48 49 PM
</details>

### CODE REVIEW

File: websites/Z/Zybooks/presence.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/* eslint-disable one-var */
+/* eslint-disable no-one-time-vars/no-one-time-vars */
```
Comment (DarkVillager) [inline websites/Z/Zybooks/presence.ts:2]: why

File: websites/Z/Zybooks/metadata.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		"id": "199605025914224641",
+		"name": "pokestarfan"
+	},
+	"service": "Zybooks",
+	"description": {
+		"en": "Share progress on Zybooks activities."
+	},
+	"url": "learn.zybooks.com",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/GV6rX6r.png",
```
Comment (DarkVillager) [inline websites/Z/Zybooks/metadata.json:13]: Logo must be 512x512 per our guideliness

File: websites/Z/Zybooks/presence.ts (hunk 1 above)
Comment (PythonCoderAS) [reply websites/Z/Zybooks/presence.ts:2]: I think one-time variables make code more readable, as it is somewhat self-documenting.

Comment (theusaf) [reply websites/Z/Zybooks/presence.ts:2]: While you and others may think that these rules make it less readable, this repository enforces these rules to be create consistent code and what the repository owners may consider as more readable. For the time being, please remove the disabling of the eslint rules. Also, thanks for your contribution!
```suggestion
```

File: websites/Z/Zybooks/metadata.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+		"id": "199605025914224641",
+		"name": "pokestarfan"
+	},
+	"service": "Zybooks",
+	"description": {
+		"en": "Share progress on Zybooks activities."
+	},
+	"url": "learn.zybooks.com",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/viwt0Wj.png",
```
Comment (theusaf) [inline websites/Z/Zybooks/metadata.json:13]: This logo appears to be stretched strangely.

File: websites/Z/Zybooks/metadata.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	"thumbnail": "https://i.imgur.com/I4RTS73.png",
```
Comment (theusaf) [inline websites/Z/Zybooks/metadata.json:14]: The thumbnail may be a bit too small/low-res. A better one could is a screenshot of the website.

File: websites/Z/Zybooks/presence.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+const presence = new Presence({
+	clientId: "1152294065115779102",
+});
+
+function getTextContent(querySelectorString: string): string {
+	return (
+		(
+			document.querySelector(querySelectorString) as HTMLElement | null
```
Comment (theusaf) [inline websites/Z/Zybooks/presence.ts:10]:
```suggestion
			document.querySelector<HTMLElement >(querySelectorString)
```

File: websites/Z/Zybooks/presence.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	hash: "",
+	timestamp: Date.now(),
+};
+
+presence.on("UpdateData", async () => {
+	const { pathname, href } = document.location;
+	let hash: string = href;
+	const presenceData: PresenceData = {
+		startTimestamp: zybooksTimestamp.timestamp,
+		largeImageKey: "https://i.imgur.com/I4RTS73.png",
```
Comment (theusaf) [inline websites/Z/Zybooks/presence.ts:28]: The logo should be a 512x512 square.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/Z/Zybooks/presence.ts (hunk 6 above)
Comment (PythonCoderAS) [reply websites/Z/Zybooks/presence.ts:26]: The logo isn't square to begin with so I don't know what to do

Comment (theusaf) [reply websites/Z/Zybooks/presence.ts:26]: There are tools that can edit images and do things like upscaling them, squaring them, etc.

File: websites/Z/Zybooks/metadata.json (hunk 3 above)
Comment (DarkVillager) [inline websites/Z/Zybooks/metadata.json:13]:
```suggestion
	"logo": "https://i.imgur.com/1lSQQh1.png",
```

File: websites/Z/Zybooks/presence.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
```
Comment (DarkVillager) [inline websites/Z/Zybooks/presence.ts:26]: Here a square 512x512 logo
```suggestion
		largeImageKey: "https://i.imgur.com/1lSQQh1.png",
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/Z/Zybooks/presence.ts (hunk 6 above)
Comment (DarkVillager) [reply websites/Z/Zybooks/presence.ts:26]: > The logo isn't square to begin with so I don't know what to do

For reference the way I do it is:
1- Download the image
2- Open it in paint.net
3- Change canvas size to 512x512 and use the image tool to place the old image in the middle
4- Delete the white parts
5- _boom_ you're doine

File: websites/Z/Zybooks/presence.ts (hunk 7 above)
Comment (PythonCoderAS) [reply websites/Z/Zybooks/presence.ts]: Thank you!

File: websites/Z/Zybooks/metadata.json (hunk 4 above)
Comment (theusaf) [reply websites/Z/Zybooks/metadata.json:14]: Ensure that this is resolved

**Review by theusaf — CHANGES_REQUESTED** [review]

**Comment by theusaf** [comment]
Closing for inactivity. You may open a new PR after resolving the issues mentioned in this one.

**Pull request closed without merging**